flink-table/flink-table-common/src/main/java/org/apache/flink/table/functions/BuiltInFunctionDefinitions.java

```diff
@@ -1504,6 +1506,23 @@
                     .runtimeDeferred()
                     .build();
 
+    public static final BuiltInFunctionDefinition JSON_VALUE =
+            BuiltInFunctionDefinition.newBuilder()
+                    .name("JSON_VALUE")
+                    .kind(SCALAR)
+                    .inputTypeStrategy(
+                            sequence(
+                                    logical(LogicalTypeFamily.CHARACTER_STRING),
+                                    and(logical(LogicalTypeFamily.CHARACTER_STRING), LITERAL),
+                                    TYPE_LITERAL,
+                                    symbol(JsonValueOnEmptyOrError.class),
+                                    ANY,
+                                    symbol(JsonValueOnEmptyOrError.class),
+                                    ANY))
+                    .outputTypeStrategy(argument(2))
```

Comment:
this would lead to NOT NULL if a user passes `DataTypes.INT().notNull()`, not sure if we want that. how about we rename `nullable()` to `nullableIfArgs`? and use `nullable(argument(2))` to make it always nullable.

Reply:
My pleasure renaming that method.